packages/logging/src/log-escalator.test.ts

```diff
@@ -0,0 +1,310 @@
+import pino from 'pino'
+import { logEscalator } from './log-escalator'
+
+it('does not write logs until done is called', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'info' },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
+    wrappedLog.fatal('Test')
+
+    expect(logs).toEqual([])
+})
+
+it('wraps a logger', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'trace', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.debug('Test')
+    wrappedLog.info('Test')
+    wrappedLog.warn('Test')
+    wrappedLog.error('Test')
+    wrappedLog.fatal('Test')
+    wrappedLog.done()
+
+    expect(logs).toEqual([
+        `{"level":10,"msg":"Test","v":1}
+`,
+        `{"level":20,"msg":"Test","v":1}
+`,
+        `{"level":30,"msg":"Test","v":1}
+`,
+        `{"level":40,"msg":"Test","v":1}
+`,
+        `{"level":50,"msg":"Test","v":1}
+`,
+        `{"level":60,"msg":"Test","v":1}
+`,
+    ])
+})
+
+it('done clears buffer', () => {
+    const logs: string[] = []
+    const log = pino(
+        { level: 'trace', timestamp: false, base: {} },
+        {
+            write(msg) {
+                logs.push(msg)
+            },
+        },
+    )
+
+    const wrappedLog = logEscalator(log)
+
+    wrappedLog.trace('Test')
+    wrappedLog.done()
+    wrappedLog.done()
```

Comment:
What's the purpose of calling done twice here?

Reply:
Ensuring that the logs are not doubled up (ie the log buffer is cleared)

Reply:
Oh ok, that makes sense.

Reply:
Can you continue to use a logEscalator after calling `.done()`? If so, it might be less confusing to call it something like `.flush()` or `.emit()` instead.

Reply:
You can continue using it, but it will no longer buffer, it will just pass straight through, which is why I was thinking done, but not really a fan of that. flush didn't seem right either. emit is ok?

Reply:
Gonna resolve this, I think it's good enough for now.